File: app/src/main/java/com/mohnage7/movies/viewmodel/MovieDetailViewModel.java
package com.mohnage7.movies.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.mohnage7.movies.MoviesApplication;
import com.mohnage7.movies.base.DataWrapper;
import com.mohnage7.movies.model.MovieVideos;
import com.mohnage7.movies.repository.MovieDetailsRepository;

import javax.inject.Inject;

public class MovieDetailViewModel extends ViewModel {

    @Inject
    MovieDetailsRepository repository;

    public MovieDetailViewModel() {
        MoviesApplication.getInstance().getDataComponent().inject(this);
    }


    public LiveData<DataWrapper<MovieVideos>> getVideosList(int movieId) {
        return repository.getVideos(movieId);
    }

}

File: app/src/main/java/com/mohnage7/movies/features/moviedetails/viewmodel/MovieDetailViewModel.kt
package com.mohnage7.movies.features.moviedetails.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.mohnage7.movies.features.moviedetails.model.MovieVideos
import com.mohnage7.movies.features.moviedetails.repository.MovieDetailsRepository
import com.mohnage7.movies.network.model.DataWrapper

class MovieDetailViewModel(private val repository: MovieDetailsRepository) : ViewModel() {

    fun getVideosList(movieId: Int): LiveData<DataWrapper<MovieVideos>> {
        return repository.getVideos(movieId)
    }
}
